gz_topic/tutorial.md

```diff
@@ -0,0 +1,206 @@
+
+which produces the following output:
+
+~~~
+/gazebo/default/atmosphere
+/gazebo/default/diagnostics
+/gazebo/default/factory
+/gazebo/default/factory/light
+/gazebo/default/gui
+/gazebo/default/joint
+/gazebo/default/light/modify
+/gazebo/default/log/control
+/gazebo/default/log/status
+/gazebo/default/model/info
+/gazebo/default/model/modify
+/gazebo/default/physics
+/gazebo/default/physics/contacts
+/gazebo/default/playback_control
+/gazebo/default/pose/info
+/gazebo/default/pose/local/info
+/gazebo/default/pose/modify
+/gazebo/default/request
+/gazebo/default/response
+/gazebo/default/undo_redo
+/gazebo/default/user_cmd
+/gazebo/default/user_cmd_stats
+/gazebo/default/visual
+/gazebo/default/wind
+/gazebo/default/world_control
+/gazebo/default/world_stats
+/gazebo/server/control
+/gazebo/world/modify
+~~~
+
+# `gz topic --info`
+
+To get more information about a particular gazebo-transport topic, use the
+`--info` or `-i` parameter.
+
+~~~
+gz topic -i /gazebo/default/world_stats
+~~~
+
+which prints the type of message published on the topic and information
+about publishers and subscribers of that topic.
+
+~~~
+Type: gazebo.msgs.WorldStatistics
+
+Publishers:
+	192.168.0.197:38057
+
+Subscribers:
+
+
+~~~
+
+# `gz topic --echo`
+
+To view the contents of messages published on a particular gazebo-transport
+topic, use the `--echo` or `-e` parameter. To echo messages for a finite
+duration of time, use the `--duration` or `-d` parameter as well.
+
+~~~
+gz topic --echo /gazebo/default/world_stats --duration 1
+~~~
+
+~~~
+sim_time {
+  sec: 656
+  nsec: 31000000
+}
+pause_time {
+  sec: 0
+  nsec: 0
+}
+real_time {
+  sec: 657
+  nsec: 883386244
+}
+paused: false
+iterations: 656031
+
+sim_time {
+  sec: 656
+  nsec: 230000000
+}
+pause_time {
+  sec: 0
+  nsec: 0
+}
+real_time {
+  sec: 658
+  nsec: 83766729
+}
+paused: false
+iterations: 656230
+
+~~~
+
+# `gz topic --hz` and `gz topic --bw`
+
+The `--echo` command can be quite verbose. If you want to see only the
+frequency with which messages are published on a topic, use the `--hz`
+or `-z` parameter. If you want to see the bandwidth used by the topic,
+use `--bw` or `-b`.
+
+~~~
+gz topic --hz /gazebo/default/world_stats
+~~~
+
+~~~
+Hz:   4.98
+Hz:   4.96
+Hz:   5.01
+Hz:   4.99
+Hz:   5.00
+Hz:   4.98
+Hz:   4.98
+Hz:   4.99
+~~~
+
+~~~
+gz topic --bw /gazebo/default/world_stats
+~~~
+
+~~~
+Total[182.85 B/s] Mean[33.36 B] Min[33 B] Max[34 B] Messages[11]
+Total[185.19 B/s] Mean[33.40 B] Min[33 B] Max[34 B] Messages[10]
+Total[185.08 B/s] Mean[33.40 B] Min[33 B] Max[34 B] Messages[10]
+Total[185.16 B/s] Mean[33.40 B] Min[33 B] Max[34 B] Messages[10]
+Total[185.02 B/s] Mean[33.40 B] Min[33 B] Max[34 B] Messages[10]
+~~~
+
+# `gz topic --pub`
+
+Added in `gazebo10` by [Bitbucket pull request 2951](https://osrf-migration.github.io/gazebo-gh-pages/#!/osrf/gazebo/pull-requests/2951/page/1),
+a message can be published from the command-line by using the `--pub`
+or `-p` parameter.
+
```

Comment:
Can you add here an example ? I can see some examples below but I think it just for consistency with the other commands above

Reply:
I added a little more description of the message syntax in 5660616 and also an additional example. I think the examples can stand in their own sub-sections since they are a little complicated